acceptance/README.md

```diff
@@ -0,0 +1,113 @@
+# Acceptance tests for puppetlabs-puppet_agent
+
+## Background
+
+### About Beaker
+
+Beaker is a host provisioning and an acceptance testing framework. If you are
+unfamiliar with beaker, you can start with these documents:
+
+- [The Beaker DSL document](https://github.com/puppetlabs/beaker/blob/master/docs/how_to/the_beaker_dsl.md) will help you understand the test code in the `tests/` and `pre_suite/` subdirectories.
+- [The Beaker Style Guide](https://github.com/puppetlabs/beaker/blob/master/docs/concepts/style_guide.md) will help you write new test code.
+- [Argument Processing](https://github.com/puppetlabs/beaker/blob/master/docs/concepts/argument_processing_and_precedence.md) and [Using Subcommands](https://github.com/puppetlabs/beaker/blob/master/docs/tutorials/subcommands.md) have more information on beaker's command line and environmental options.
+
+### About these tests
+
+Testing this module necessarily involves repeatedly installing and
+uninstalling puppet-agent. Ideally, the test hosts would be totally destroyed
+and reprovisioned before each fresh install of puppet-agent, but beaker does not
+support workflows like this. Instead:
+
+- Use the `run_setup` helper method to install puppet-agent, plus this module
+  and its dependencies, before the main content of each of your tests.
+- Use the `run_teardown` helper method to uninstall puppet-agent and remove
+  modules in your tests' teardown steps.
+
+See [helpers.rb](./helpers.rb) for the impelementations of these methods.
+
+#### Environment variables affecting test behavior
```

Comment:
The list just below this heading explains how the version is selected. If X_AGENT_VERSION is found, it will use the specific version. If that's not set, it will use X_PUPPET_COLLECTION. If that's not set, it will use the PC1 collection.

Reply:
Oh I think I see what you're saying. If both a version and a collection are supplied, the collection will be ignored. It doesn't make sense to have both control the installation.

Reply:
To clarify, you're saying that if I supply `FROM_AGENT_VERSION` and `FROM_PUPPET_COLLECTION`, then `FROM_AGENT_VERSION` will be used, right?

What about if I supply a version and a collection (in different directions, so e.g. `FROM_AGENT_VERSION` and `TO_PUPPET_COLLECTION`) [unless that's what you were referring to]

Reply:
It follows the same process to decide on the "to" agent as it does to decide on the "from" agent -- the two sets of variables don't depend on one another. Do you have suggestions about how to improve this wording, or about how the variables should behave or be named?

Reply:
I should say that that's the _desired_ behavior, though! how puppet collections behave in the "to" context will have to be managed by each test individually, or by other, new helpers, later

Reply:
For example, in a test of the `package_version` parameter, it wouldn't make much sense to use a puppet collection. Perhaps a test that tests the collection could use a helper from beaker-puppet to find the collection based on the TO_AGENT_VERSION if TO_PUPPET_COLLECTION wasn't set